### PR TITLE
feat(clone): add clone

### PR DESCRIPTION
**Pull request by juno7803**
Hello, @raon0211! 😁 I have implemented a clone function with shallow copy. However, I have some questions!

image

### Why `clone` function in es-toolkit targets only plain objects and plain arrays? 🤔
By plain objects, do you mean objects that are pure key-value pairs, excluding Date, Regex, Map, etc.? If so, I'm curious why you requested a clone that targets only plain objects and plain arrays in the issue.

Since this library is intended to target `lodash`, I wondered if it should support other types of data structures as well.

For now, I've added support for other data types, but I can change this based on your response :)

### Implementation of cloneDeep
This is not directly related to clone, but I have a question since it overlaps with the topic of plain objects and plain arrays.

The relatively recent API, [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone), supports deep cloning. It is available in most modern browsers except IE, and there are quite a few libraries written using this API ([remeda](https://remedajs.com/docs/#clone), [atomic-fns](https://atomic-fns.dev/functions/collections.clone.html)).

image

Since es-toolkit is a modern JavaScript library, I think it would be good to use this API. However, if we are only targeting plain objects and plain arrays, I'm wondering if there is a need to implement cloneDeep separately.

On the other hand, structuredClone throws an error when used with unsupported data types, so it might be worthwhile to provide deepClone to handle these cases!

Please read my thoughts and give me your feedback, and I'll adjust the pull request accordingly. Thank you :)

### CODE REVIEW

**Comment by bovas85** [comment]
I feel a like-for-like cloneDeep (and clone) would be beneficial as the main use-case for users is to replace lodash functions here

File: docs/ko/reference/object/clone.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+console.log(clonedNum === num); // true
+
+const arr = [1, 2, 3];
+const clonedArr = clone(arr);
+console.log(clonedArr); // [1, 2, 3]
+console.log(clonedArr === arr); // false
+
+const obj = { a: 1, b: 'es-toolkit', c: [1, 2, 3] };
+const clonedObj = clone(obj);
+console.log(clonedObj); // { a: 1, b: 'es-toolkit' }
```
Comment (raon0211) [inline docs/ko/reference/object/clone.md:34]:
```suggestion
console.log(clonedObj); // { a: 1, b: 'es-toolkit', c: [1, 2, 3] }
```

**Comment by raon0211** [comment]
I thought supporting plain objects is usually the most straightforward way, but I guess we might support native JavaScript objects, too. I will also add support for `Map`s and `Set`s.

**Comment by raon0211** [comment]
When it comes to `cloneDeep`, I agree that we use `structuredClone` to simplify our codebase!

**Review by raon0211 — APPROVED** [review]
Thanks for your great contribution!

**Comment by juno7803** [comment]
@raon0211 I agree with your feedbacks! Based on your feedback, I will also implement cloneDeep. Thank you! 😊

**Pull request merged**